src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -870,6 +899,26 @@ void StorageKafka::updateConfiguration(cppkafka::Configuration & kafka_config)
     }
 }
 
+void StorageKafka::updateConsumerConfiguration(cppkafka::Configuration & kafka_config)
+{
+    const auto & config = getContext()->getConfigRef();
+    auto config_prefix = getConfigPrefix();
+    if (config.has(config_prefix))
+        loadConsumerConfig(kafka_config, config, config_prefix, topics);
+
+    loadTopicLegacyConfig(kafka_config, config, config_prefix, topics);
```

Comment:
Should this be inside `loadConsumerConfig`? Since without it will not have `consumer` in the config prefix and will load nothing I guess (same for producer)

Reply:
Yes, makes sense.